addons/l10n_account_withholding/tests/test_account_withholding.py

```diff
@@ -0,0 +1,789 @@
+
+    def _setup_tax(self, name, amount, sequence=None, tax_type='sale', price_include='tax_included', base_tag=None, tax_tag=None):
+        # Copy a default tax and set it up for withholding
+        tax = self.company_data['default_tax_sale'].copy({
+            'name': name,
+            'amount': amount,
+            'type_tax_use': tax_type,
+            'price_include_override': price_include,
+            'is_withholding_tax_on_payment': True,
+            'withholding_sequence_id': sequence and sequence.id,
+        })
+        # Add tax grids
+        tax.invoice_repartition_line_ids.filtered(lambda x: x.repartition_type == 'tax').tag_ids = tax_tag or self.env['account.account.tag'].create({
+            'name': f'Tax Tag {name}',
+            'applicability': 'taxes',
+        })
+        tax.invoice_repartition_line_ids.filtered(lambda x: x.repartition_type == 'base').tag_ids = base_tag or self.env['account.account.tag'].create({
+            'name': f'Base Tag {name}',
+            'applicability': 'taxes',
+        })
+        return tax
+
+    def _get_tax_tag(self, tax):
+        return {
+            'tax': tax.invoice_repartition_line_ids.filtered(lambda x: x.repartition_type == 'tax').tag_ids.ids,
+            'base': tax.invoice_repartition_line_ids.filtered(lambda x: x.repartition_type == 'base').tag_ids.ids,
+        }
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        # Set the withholding account
+        cls.company_data['company'].withholding_tax_base_account_id = cls.env['account.account'].create({
+            'code': 'WITHB',
+            'name': 'Withholding Tax Base Account',
+            'reconcile': True,
+            'account_type': 'asset_current',
+        })
+        # Create a sequence to set on tax C
+        # Note that we use no gap for the sake of testing, in order to have the sequences be rollbacked between tests.
+        cls.withholding_sequence = cls.env['ir.sequence'].create({
+            'implementation': 'no_gap',
+            'name': 'Withholding Sequence',
+            'padding': 4,
+            'number_increment': 1,
+        })
+        # Prepare two withholding taxes.
+        cls.tax_sale_b = cls._setup_tax(cls, 'Withholding Tax', 1)
+        cls.tax_sale_c = cls._setup_tax(cls, 'Withholding Tax 2', 2, cls.withholding_sequence)
+        # Prepare one draft invoice for the tests.
+        cls.invoice = cls.env['account.move'].create({
+            'move_type': 'out_invoice',
+            'date': '2024-01-01',
+            'invoice_date': '2024-01-01',
+            'partner_id': cls.partner_a.id,
+            'invoice_line_ids': [Command.create({'product_id': cls.product_a.id, 'price_unit': 1000.0})],
+        })
+        # Set a withholding tax on product B to test later.
+        cls.product_b.taxes_id = cls.tax_sale_b
+        # We'll need a foreign currency
+        cls.foreign_currency = cls.setup_other_currency('EUR')
```

Comment:
no rate?

Reply:
I'm just using the default rate, works well enough